packages/relay/src/lib/eth.ts

```diff
@@ -1578,46 +1583,24 @@ export class EthImpl implements Eth {
         );
 
       txSubmitted = true;
-      fileId = sendRawTransactionResult!.fileId;
-
-      // Wait for the record from the execution.
-      const txId = sendRawTransactionResult!.txResponse.transactionId.toString();
-      const formattedId = formatTransactionIdWithoutQueryParams(txId);
-
-      // handle formattedId being null
-      if (!formattedId) {
-        throw predefined.INTERNAL_ERROR(`Invalid transactionID: ${txId}`);
-      }
-
-      const contractResult = await this.mirrorNodeClient.repeatedRequest(
-        this.mirrorNodeClient.getContractResult.name,
-        [formattedId, requestDetails],
-        this.mirrorNodeClient.getMirrorNodeRequestRetryCount(),
-        requestDetails,
-      );
-
-      if (!contractResult) {
-        this.logger.warn(`${requestIdPrefix} No record retrieved`);
-        throw predefined.INTERNAL_ERROR(`No matching record found for transaction id ${txId}`);
-      }
-
-      if (contractResult.hash == null) {
-        this.logger.error(
-          `${requestIdPrefix} The ethereumHash can never be null for an ethereum transaction, and yet it was!!`,
+      fileId = sendRawTransactionResult.fileId;
```

Comment:
@quiet-node So with this change won't the fileId be null when the _**executeTransaction**_ in the **_submitTransaction_** throws null? And this will result in the file not being deleted?

Reply:
Yes, you're correct! However, the original implementation has a flaw: if an error is thrown, `fileId` does not get updated and thus eventually doesn't get deleted. The previous implementation didn't fully address this issue; it allowed Connection Dropped and Timeout errors to carry the created `fileId` but didn't account for other errors. I've noted this and captured it in this ticket https://github.com/hashgraph/hedera-json-rpc-relay/issues/3219, which we can triage and address in the next sprint.